src/__tests__/api.test.ts

```diff
@@ -0,0 +1,29 @@
+import { buildDecoratorFactories, FACTORIES, PropertyDecoratorFactory } from '..';
+
+describe('buildDecoratorFactories', () => {
+    it('produces new decorator factories', () => {
+        const noop = () => { };
+        const transformer = () => noop;
+
+        const factories = buildDecoratorFactories({
```

Comment:
An actual implementation would write a different transformer for each decorator name.